server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -124,8 +123,7 @@
             SETTING_NUMBER_OF_SHARDS,
             SETTING_VERSION_CREATED,
             SETTING_INDEX_UUID,
-            SETTING_CREATION_DATE,
-            IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey()));
```

Comment:
why remove this? We still don't want people to be able to modify this on restore, do we? Otherwise they could just set this to false on restore.

Reply:
fixed in https://github.com/elastic/elasticsearch/pull/51122/commits/9a992a9aff741adc98e494eed9814769e7e21dbc